### PR TITLE
 	Bug 1186558 - Upgrade bootstrap to v3.3.5

### PR DESCRIPTION
**Pull request by wlach**
[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/801)

### CODE REVIEW

File: ui/vendor/css/bootstrap.css [hunk 1]
```diff
@@ -4087,7 +4527,16 @@ select[multiple].input-group-sm > .input-group-btn > .btn {
...
   }
-  .navbar-text.navbar-right:last-child {
+}
+@media (min-width: 768px) {
+  .navbar-left {
+    float: left !important;
+  }
+  .navbar-right {
+    float: right !important;
+    margin-right: -15px;
```
Comment (tojonmz) [inline ui/vendor/css/bootstrap.css]: So I think this new `margin-right: -15px;` is causing our upper right navbars to shrink, including the quick filter. We may need to add our own treeherder.css override with a `margin-right: 0;` if we want to preserve our current navbar appearance and quick filter dimension.

The default bootstrap appearance showing the shrinkage
navbarright

File: ui/vendor/css/bootstrap.css [hunk 2]
```diff
@@ -910,12 +1097,12 @@ textarea {
   line-height: inherit;
 }
 a {
-  color: #428bca;
+  color: #337ab7;
```
Comment (tojonmz) [inline ui/vendor/css/bootstrap.css]: So they've also changed link colors everywhere, if we care about that. Darker and less saturated blue.

links

File: ui/vendor/css/bootstrap.css [hunk 3]
```diff
@@ -2423,12 +2633,12 @@ input[type="month"].input-lg {
 .checkbox {
   position: relative;
   display: block;
-  min-height: 20px;
```
Comment (tojonmz) [inline ui/vendor/css/bootstrap.css]: Fwiw, this removal of `min-height` means we've lost our min height on our Similar Jobs checkboxes per screen grab below. They are up against the bottom of the grey container in stead of having 2px of breathing space below them.

checkboxminheight

File: ui/vendor/css/bootstrap.css (hunk 1 above)
Comment (tojonmz) [inline ui/vendor/css/bootstrap.css]: Ya, we'll have to add that change above. Without it the quick filter clear [x] icon is off screen when a filter is active:

filterclipped

File: ui/vendor/css/bootstrap.css [hunk 4]
```diff
@@ -264,7 +264,7 @@ th {
   font-family: 'Glyphicons Halflings';
 
   src: url('../fonts/glyphicons-halflings-regular.eot');
-  src: url('../fonts/glyphicons-halflings-regular.eot?#iefix') format('embedded-opentype'), url('../fonts/glyphicons-halflings-regular.woff') format('woff'), url('../fonts/glyphicons-halflings-regular.ttf') format('truetype'), url('../fonts/glyphicons-halflings-regular.svg#glyphicons_halflingsregular') format('svg');
+  src: url('../fonts/glyphicons-halflings-regular.eot?#iefix') format('embedded-opentype'), url('../fonts/glyphicons-halflings-regular.woff2') format('woff2'), url('../fonts/glyphicons-halflings-regular.woff') format('woff'), url('../fonts/glyphicons-halflings-regular.ttf') format('truetype'), url('../fonts/glyphicons-halflings-regular.svg#glyphicons_halflingsregular') format('svg');
```
Comment (tojonmz) [inline ui/vendor/css/bootstrap.css]: I think we need to add the `glyphicons-halflings-regular.woff2` resource, per console
https://github.com/mozilla/treeherder/blob/master/ui/vendor/fonts/

glyphiconerrors

**Comment by tojonmz** [comment]
Other than comments in the changes, in local testing Perfherder shows various extra whitespace (which isn't a bad thing). I haven't sussed out where it's coming from though. Perhaps grunt build whitespace collapse on deployment? Or part of this bootstrap change.

Local vagrant appearance
perfchanges

Other than the other comments and issues, I think we are good.

**Comment by wlach** [comment]
I made some minor modifications to `treeherder.css` to account for the layout problems you noticed.

I decided we could live with the slightly different link color. You really have to stare to notice the difference. :)

The perfherder whitespace stuff was fixed by using conservativeCollapse in bug 1187035.

Going to merge this.

**Pull request merged**